### PR TITLE
Fixup ci action

### PR DESCRIPTION
**Pull request by cpuguy83**
1. Fix a typo in envtest name
2. Add caches for go artifacts
3. Add caches for fetched e2e binaries

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -40,11 +40,19 @@ jobs:
         with:
           go-version: ${{ env.GO_VERSION }}
       - uses: actions/checkout@v2
```
Comment (helayoty) [inline .github/workflows/ci.yml:42]: Remove the duplicate checkout step.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -54,6 +62,14 @@ jobs:
...
```
Comment (helayoty) [inline .github/workflows/ci.yml:64]: Remove the duplicate checkout step.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -77,12 +93,20 @@ jobs:
           go-version: ${{ env.GO_VERSION }}
       - name: Checkout repository
```
Comment (helayoty) [inline .github/workflows/ci.yml:94]: Remove the duplicate checkout step.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -77,12 +93,20 @@ jobs:
...
         uses: actions/checkout@v3
```
Comment (helayoty) [inline .github/workflows/ci.yml:95]: Remove thew duplicate checkout step.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -106,5 +143,13 @@ jobs:
...
             sleep 1;
           done
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-e2e-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-e2e
```
Comment (helayoty) [inline .github/workflows/ci.yml:153]: Why do we need to have the checkout step again here?

**Comment by cpuguy83** [comment]
Gonna close this out.
Some of the changes I already merged into main.
The other parts of this was to get github actions cache working, but it gets throttled even with just a few runs... so more trouble than it is worth.

**Pull request closed without merging**